deployment/7_upgrade_v1_1.js

```diff
@@ -109,16 +116,22 @@ const deploy = async (network) => {
             for (let i = 0; i < toRemove.length; i++) {
                 await multisigExecutor.executeCall(ModuleRegistryWrapper, "deregisterModule", [toRemove[i].address]);
             }
+        } else {
+            // add all modules present in newVersion that are not present in version
+            toAdd = newVersion.modules.filter(module => !version.modules.map(m => m.address).includes(module.address));
+            // remove all modules from version that are no longer present in newVersion
+            toRemove = version.modules.filter(module => !newVersion.modules.map(m => m.address).includes(module.address));
         }
-        
+
         const UpgraderWrapper = await deployer.deploy(
             Upgrader,
             {},
-            toRemove.map((module) => {return module.address;}),
-            toAdd.map((module) => {return module.address;})
+            config.contracts.ModuleRegistry,
+            toRemove.map(module => module.address),
+            toAdd.map(module => module.address)
         );
-        const upgraderName = version.fingerprint + '_' + fingerprint; 
-        await multisigExecutor.executeCall(ModuleRegistryWrapper, "registerUpgrader", [UpgraderWrapper.contractAddress, utils.asciiToBytes32(upgraderName)]);
+        const upgraderName = version.fingerprint + '_' + fingerprint;
+        await multisigExecutor.executeCall(ModuleRegistryWrapper, "registerModule", [UpgraderWrapper.contractAddress, utils.asciiToBytes32(upgraderName)]);
```

Comment:
Following my earlier comment about keeping the current ModuleRegistry, we need to make sure that adding the upgrader triggers the `UpgraderRegistered` event so we should register the upgrader as module and as an upgrader.